overlord/hookstate/ctlcmd/ctlcmd_test.go

```diff
@@ -112,3 +113,35 @@ func (s *ctlcmdSuite) TestHiddenCommand(c *C) {
 	// mock-hidden is not in the help message
 	c.Check(err.Error(), Not(testutil.Contains), "  mock-hidden\n")
 }
+
+func (s *ctlcmdSuite) TestRootRequiredCommandFailure(c *C) {
+	_, _, err := ctlcmd.Run(s.mockContext, []string{"start"}, 1000)
+
+	c.Check(err, FitsTypeOf, &ctlcmd.ForbiddenCommandError{})
+	c.Check(err.Error(), Equals, `cannot use "start" with uid 1000, try with sudo`)
+}
```

Comment:
can you also add (possibly in a followup) a test case with uid 1000 where a only-root-allowed command (such as set) is the first argument, and some non-root-allowed command is later in the arguments such as in `snapctl set get foo`.

This will test that we are specifically only checking the first argument as the command to see if it is allowed. I can foresee a security issue if we change the `strutil.ListContains(nonRootAllowed, args[0])` check to something that allows any item in the args list to match with `nonRootAllowed`. It would be a weird refactor so is probably unlikely but I think one more unit test does not hurt us.

But also okay for a followup too.

Reply:
Definitely, doesn't hurt to be defensive. I added the test